src/gui/src/ScreenSetupView.cpp

```diff
@@ -65,18 +66,82 @@ void ScreenSetupView::resizeEvent(QResizeEvent* event)
     event->ignore();
 }
 
+void ScreenSetupView::enter(const QModelIndex& index)
+{
+    if (!index.isValid())
+        return;
+    Screen& screen = model()->screen(index);
+    if (screen.isNull())
+        screen = Screen(tr("Unnamed"));
+    ScreenSettingsDialog dlg(this, &screen);
+    dlg.exec();
+}
+
+void ScreenSetupView::remove(const QModelIndex& index)
+{
+    if (!index.isValid())
+        return;
+    Screen& screen = model()->screen(index);
+    if (!screen.isNull()) {
+        screen = Screen();
+        emit dataChanged(index, index);
+    }
+}
+
+void ScreenSetupView::keyPressEvent(QKeyEvent* event)
+{
+    if (event->key() == Qt::Key_Return || event->key() == Qt::Key_Delete)
+    {
+        QModelIndexList indexes = selectedIndexes();
+        if (indexes.count() == 1 && indexes[0].isValid())
+        {
+            if (event->key() == Qt::Key_Return)
+                enter(indexes[0]);
+            else if (event->key() == Qt::Key_Delete)
+                remove(indexes[0]);
+        }
+        // Do not let base handle the event, at least not for return key because it
+        // results in next esc/return key in the opened Screen Settings dialog not
+        // only closing that but also the parent Server Configuration dialog.
+    }
+    else if ((event->modifiers() & Qt::ControlModifier)
+        && (event->key() == Qt::Key_Left || event->key() == Qt::Key_Right || event->key() == Qt::Key_Up || event->key() == Qt::Key_Down))
```

Comment:
Still too long, let's keep the length at no more than 100 characters. Sorry for not being explicit the first time.

Reply:
Sure. Done. (Considered it myself, but did not because you said two and also unsure about what is most readable with such && .. || .. combinations.)